aws-android-sdk-appsync-tests/src/androidTest/java/com/amazonaws/mobileconnectors/appsync/AppSyncTestSetupHelper.java

```diff
@@ -101,74 +148,54 @@ public void onFailure(PersistentMutationsError error) {
                             Log.e(TAG, "Error", error.getException());
                         }
                     })
+                    .useClientDatabasePrefix(true)
```

Comment:
This overall idea of using a "builder" style pattern with direct specification of configuration intermixed with an AWSConfiguration.json that also has configuration is confusing. It is unclear to me why the prefix would be specified in the awsconfiguration.json file, but the flag to use the prefix would be specified in the builder.

Reply:
This is the design we agreed upon and here are the reasons to it:-
1) Isolate Multi-Auth changes for customers who want it: For a developer who is not interested in Multi-Auth and who upgrades the SDK, we do not want to migrate their databases or introduce a breaking change and just let the `AWSAppSyncClient` object use the default database as is.

2) We want the customers to explicitly say that they want to use database prefix for the database name through a flag.

3) Convention over configuration: Our convention is to use `awsconfiguration.json`. So we are adding the client database prefix in the file and the flag indicates if the prefix should be used.